### DISCUSSION TITLE
End of year 2021: 20 months of Gitoxide

### DISCUSSION THREAD

**Discussion by Byron** [GitoxideLabs/gitoxide#285] [General]
###### The Beginning

Despite `gitoxide` being created mid 2018, it really only started active development in April **2020 at 1399 SLOC and 1758 lines total** in **60 commits**, just **2 crates** with the main binary being called `grit`. There was just a single author, me. On GitHub, there were **14 stars**. A lot would change in 2020… .

###### 2020

At the end of 2020, the numbers look quite different with **21009 SLOC and 24749 lines total** in **1847 commits**. There were **14 crates** and the main binaries were now called `gix` and `gixp` (for plumbing). In total there were **5 unique authors**, one of which wrote an entire crate, `git-commitgraph`. In the 263 days of that year, that would mean 87~ lines written each day in 7 commits each. On GitHub, there were **1699 stars**, or ~6.6 stars per day. Despite it picked up steam, 2021 would change the pace once again… .

###### 2021

And 361 days later as of 2021-12-28, we are counting **62138 SLOC and 87340 lines total** in **5671 commits**. There are **27 crates** and 3 binaries, `ein` (formerly `gix`), `gix` (formerly `gixp`), and `cargo smart-release/changelog`. There are **21 unique authors**, with one contributing the entire `git-config` crate. This means ~173 lines per day in ~11 commits each. On GitHub there are **2779 stars** for ~3 stars per day.

The tool invocation `ein tools estimate-hours` now rates the project cost at **3623 hours or ~453 in 8 hour working days**, for an average working time of **5.8 hours in the past 624 days**.

###### The Future…

Making a prediction for 2022 would be fun for sure, but I'd rather hold back to not accidentally set unrealistic expectations while hoping to keep up this pace, thinking even 5h effective per day sustained would be phenomenal.

###### Thanks and Wishes

**Let me use this opportunity to thank everybody involved**, contributors, sponsors and supporters alike. It's a huge task to take on and each of you helped tremendously. Let me highlight that kind words in constructive interactions are the probably most underrated way of support, and they do help to keep going at all while facing the hardships that we all share to some extend.

**All in all I am looking optimistically into 2022 and wish everyone a great start into the new year**.

**Comment by trivigy** [comment]
Have been following the development for a while and truly wish the project much success. Amazing work thus far and looking forward for what is to come in 2022.
